pyiceberg/table/__init__.py

```diff
@@ -987,6 +993,7 @@ def overwrite(
         self,
         df: pa.Table,
         overwrite_filter: Union[BooleanExpression, str] = ALWAYS_TRUE,
+        case_sensitive: bool = True,
```

Comment:
Same here, let's add this to the end to avoid breaking folks who use positional arguments to pass in properties.